modules/openapi-generator/src/main/java/org/openapitools/codegen/IJsonSchemaValidationProperties.java

```diff
@@ -213,4 +217,47 @@ default void setTypeProperties(Schema p) {
             setIsAnyType(true);
         }
     }
+
+    /**
+     * @return basic type - no generics supported.
+     */
+    default String getBaseType() {
+        return null;
+    };
+
+    /**
+     * @return complex type that can contain type parameters - like {@code List<Items>} for Java
+     */
+    default String getComplexType() {
+        return getBaseType();
+    };
+
+    /**
+     * Recursively collect all necessary imports to include so that the type may be resolved.
+     *
+     * @param includeContainerTypes whether or not to include the container types in the returned imports.
+     * @return all of the imports
+     */
+    default Set<String> getImports(boolean includeContainerTypes) {
+        Set<String> imports = new HashSet<>();
+        if (includeContainerTypes || !(this.getIsArray() || this.getIsMap())) {
+            if (this.getComposedSchemas() != null) {
```

Comment:
How about moving the composed schema handling as an if statement adjacent to this if?
Composedschemas do not only exist in container types.
They can exist in type: string/object/array/unset schemas too.

Reply:
Thanks!